java/idl2jni/codegen/im_java.cpp

```diff
@@ -781,7 +814,7 @@ void writeUnionDefaultValue(ostream &os, AST_Expression::ExprType udisc_type,
     os << dv.u.ulonglong_val;
     break;
   default:
-    cerr << "ERROR: Bad discriminant type (shouldn't happen here)\n";
+    cerr << "ERROR - " << __FILE__ << ":" << __LINE__ << " - Bad discriminant type '" << udisc_type << "' (shouldn't happen here)\n";
```

Comment:
Can you have it do a `BE_arbort()`` here and in the other errors? It really shouldn't keep going in these cases.